ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -134,6 +140,26 @@ public static boolean isSystemColumn(final ColumnName columnName, final KsqlConf
     return isSystemColumn(columnName, getPseudoColumnVersionFromConfig(ksqlConfig));
   }
 
+  public static int getPseudoColumnVersionFromConfig(final KsqlConfig ksqlConfig) {
+    return ksqlConfig.getBoolean(KsqlConfig.KSQL_ROWPARTITION_ROWOFFSET_ENABLED)
+        ? CURRENT_PSEUDOCOLUMN_VERSION_NUMBER
+        : LEGACY_PSEUDOCOLUMN_VERSION_NUMBER;
+  }
+
+  public static boolean mustBeMaterializedForTableJoins(final ColumnName columnName) {
+    return MUST_BE_MATERIALIZED_FOR_TABLE_JOINS.contains(columnName);
+  }
+
+  public static boolean disallowedForPullQueries(
+      final ColumnName columnName,
+      final int pseudoColumnVersion
+  ) {
+    // option 2: make a set that is the union of DISALLOWED and systemcols,
```

Comment:
while it's a draft PR I left these in (also a todo later in the file). As commented I originally used the intersection (not union, sorry) of `DISALLOWED_FOR_PULL_QUERIES` and `getSystemColumnNames(pseudoColumnVersion)`, which ultimately felt less readable and is less performant if I create the set each time. I prefer the way written, but I figured I'd mention it here.